exploration/text-vs-code.md

```diff
@@ -17,49 +19,361 @@ Status: **Proposed**
+4. Trim all unquoted whitespace, but do not permit quoting non-simple patterns
+5. Selectively trim patterns (all whitespace is otherwise significant)
+
+### Always Quote
+
+Pros:
+- The boundary between pattern and code is always clear.
+- The quoting reduces the number of in-pattern escapes to the open/close sequence.
+  and the placeholder sequence sigils.
+- Since the pattern is already quoted, translators never have to add pattern quotes
+  in order to add PEWs to a given pattern.
+  This also might avoid some tools forcing escaping on added quotes that are needed.
+
+Cons:
+- Requires matching open/close quotes.
+
+### Never Quote Patterns
+
+In this alternative, all non-code whitespace is significant. 
+We have to use a slightly different syntax in the example, so that
+the boundary between code and pattern works.
+>```
+>{{
+>   match {$var}
+>   {when *} This pattern has a space in front (it's between \} and This)
+>   {when other}
+>      This pattern has a newline and six spaces in front of it
+>   {when moo}This pattern has no spaces in front of it, but an invisible space at the end
+>}}
+>```
+
+Pros:
+- WYSIWYG (on steroids)
+
+Cons:
+- Probably not a serious alternative: the example
+  includes any number of obvious footguns that have to be addressed
+
+### Permit pattern quoting
+
+In this alternative, non-simple patterns are trimmed, but it is 
+possible to use quoting to separate the pattern from code (and prevent trimming)
+>```
+>{match {$var}}
+>{when   0} This has no space in front of it.
+>{when one}
+>    This has no space or newline in front of it.
+>{when few}
+>   {{ This has one space at the start and the end. }}
+>{when many}   {{ This also has one space start and end. }}
+>{when *}{{You can quote patterns even without whitespace.}}
+>```
+
+Pros:
+- Code is special instead of text.
+- Easy to use (best of both worlds?)
+
+Cons:
+- Requires one of the alternate syntaxes
+- Has two ways to represent a pattern.
+- May be difficult for translators to add quotes when needed.
+
+### Trim All Unquoted
+
+In this alternative, all non-code whitespace is trimmed
+and we do not allow/provide for pattern quoting.
+Instead, PEWS whitespace must be individually quoted.
+
+> [!NOTE]
+> Whitespace quoting also works in the preceeding alternatives
+> because it is an inherent part of the syntax.
+> We don't show it in those alternatives because it is
+> distracting.
+
+>```
+>{match {$var}}
+>{when   0} This has no space in front of it.
+>{when one}
+>    This has no space or newline in front of it.
+>{when few}
+>   {||} This has one space at the start and the end. {||}
+>{when many}   {| |}This also has one space start and end.{| |}
+>{when *}
+>
+>       No amount of whitespace matters before this pattern
+>       but all of the whitespace at the end does.
+>
+>       {||}
+>```
+
+Pros:
+- Code is special, whitespace is not.
+- Makes PEWS into a "special event", alerting developers to the non-I18N aspects of it?
+
```

Comment:
```suggestion
- Avoids as many escape sequences as possible,
  as `}` does not need escaping in patterns.
  
```

Reply:
It needs escaping if it is used as ending of the "wrapping" of the string